src/index.js

```diff
@@ -4,189 +4,150 @@
-}
 
-// create a public key protobuf to be base64 string stored in config
-function keyMarshal (data, type) {
-  const RSA = 0
-
-  let epb
-  if (type === 'Public') {
-    epb = pbCrypto.PublicKey.encode({
-      Type: RSA,
-      Data: data
-    })
+  // encode/decode functions
+  toHexString () {
+    return mh.toHexString(this.id)
   }
 
-  if (type === 'Private') {
-    epb = pbCrypto.PrivateKey.encode({
-      Type: RSA,
-      Data: data
-    })
+  toBytes () {
+    return this.id
   }
 
-  return epb
+  toB58String () {
+    return mh.toB58String(this.id)
+  }
 }
 
-// this returns a base64 encoded protobuf of the public key
-function formatKey (key, type) {
-  // create der buffer of public key asn.1 object
-  const der = forge.asn1.toDer(key)
-
-  // create forge buffer of der public key buffer
-  const fDerBuf = forge.util.createBuffer(der.data, 'binary')
-
-  // convert forge buffer to node buffer public key
-  const nDerBuf = new Buffer(fDerBuf.getBytes(), 'binary')
-
-  // protobuf the new DER bytes to the PublicKey Data: field
-  const marsheledKey = keyMarshal(nDerBuf, type)
-
-  // encode the protobuf public key to base64 string
-  const b64 = marsheledKey.toString('base64')
-  return b64
-}
+exports = module.exports = PeerId
+exports.Buffer = Buffer
 
 // generation
 exports.create = function (opts) {
   opts = opts || {}
   opts.bits = opts.bits || 2048
 
-  // generate keys
-  const pair = forge.rsa.generateKeyPair({
-    bits: opts.bits,
-    e: 0x10001
-  })
-
-  // return the RSA public/private key to asn1 object
-  const asnPub = forge.pki.publicKeyToAsn1(pair.publicKey)
-  const asnPriv = forge.pki.privateKeyToAsn1(pair.privateKey)
-
-  // format the keys to protobuf base64 encoded string
-  const protoPublic64 = formatKey(asnPub, 'Public')
-  const protoPrivate64 = formatKey(asnPriv, 'Private')
-
-  // store the keys as a buffer
-  const bufProtoPub64 = new Buffer(protoPublic64, 'base64')
-  const bufProtoPriv64 = new Buffer(protoPrivate64, 'base64')
-
-  const mhId = multihashing(new Buffer(protoPublic64, 'base64'), 'sha2-256')
+  const privKey = crypto.generateKeyPair('RSA', opts.bits)
 
-  return new PeerId(mhId, bufProtoPriv64, bufProtoPub64)
+  return new PeerId(privKey.public.hash(), privKey)
 }
 
 exports.createFromHexString = function (str) {
-  return new PeerId(new Buffer(str, 'hex'))
+  return new PeerId(mh.fromHexString(str))
 }
 
 exports.createFromBytes = function (buf) {
   return new PeerId(buf)
 }
 
 exports.createFromB58String = function (str) {
-  return new PeerId(new Buffer(base58.decode(str)))
+  return new PeerId(mh.fromB58String(str))
 }
 
 // Public Key input will be a buffer
-exports.createFromPubKey = function (pubKey) {
-  const buf = new Buffer(pubKey, 'base64')
-  const mhId = multihashing(buf, 'sha2-256')
-  return new PeerId(mhId, null, pubKey)
+exports.createFromPubKey = function (key) {
+  let buf = key
+  if (typeof buf === 'string') {
+    buf = new Buffer(key, 'base64')
```

Comment:
If we settle that keys will be serialised to base64, we might as well change the .toJSON to also base64 then, instead of hex